### PR TITLE
server/schedule: use ChangePeerV2 to demote single voter directly

### PR DESCRIPTION
**Pull request by w41ter-l**
### What problem does this PR solve?

ref #4444.
ref https://github.com/tikv/tikv/issues/11463.

### What is changed and how it works?

 Change the field `ChangePeer` to `ChangePeerV2` if demoting a voter directlry. This can avoid breaking compatibility.

### Check List

Tests

- No code

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
Fix the problem that demoting a follower to learner isn't working.
```

### CODE REVIEW

**Comment by PatrickNicholas** [comment]
In addition to this PR, please also help me to review : https://github.com/pingcap/kvproto/pull/839.

**Comment by HunDunDM** [comment]
Since `ChangePeerV2` is used, `useJointConsensus == true` is required. This shows that the judgment condition of `useJointConsensus` and the logic of `buildStepsWithJointConsensus` need to be changed. At the same time, `DemoteFollower` will not be used in `buildStepsWithoutJointConsensus`, it must be split into `RemovePeer` and `AddLearner`.

**Comment by HunDunDM** [comment]
We need a new way of implementation.

**Comment by nolouch** [comment]
ptal @HunDunDM

File: server/schedule/operator/step.go [hunk 1]
```diff
@@ -409,6 +409,20 @@ func (df DemoteFollower) CheckInProgress(cluster opt.Cluster, region *core.Regio
 // Influence calculates the store difference that current step makes.
 func (df DemoteFollower) Influence(opInfluence OpInfluence, region *core.RegionInfo) {}
 
+// GetRequest get the ChangePeerV2 request
+func (df DemoteFollower) GetRequest() *pdpb.ChangePeerV2 {
```
Comment (bufferflies) [inline server/schedule/operator/step.go:413]: how about add unit test?

**Comment by HunDunDM** [comment]
#4444 is implemented in #4534, maybe we can close this PR? If you have any questions about #4534, please ask.

**Pull request closed without merging**